pkg/sources/sources.go

```diff
@@ -222,10 +232,48 @@ func mergeEventRoutes(resource string, sources config.IndexableMap[config.Source
 				out[e] = append(out[e], route)
 			}
 		}
+
+		// add routes related to recommendations
+		resForRecomms := recommendation.ResourceEventsForConfig(srcGroupCfg.Kubernetes.Recommendations)
+		setEventRouteForRecommendationsIfShould(&out, resForRecomms, srcGroupName, resource)
 	}
+
 	return out
 }
 
+func setEventRouteForRecommendationsIfShould(routeMap *map[config.EventType][]route, resForRecomms map[string]config.EventType, srcGroupName, resourceName string) {
+	if routeMap == nil {
+		return
+	}
```

Comment:
This map shouldn't be nil, but it's a good practice to ensure if a pointer is not nil until you access its value to avoid panics.

Theoretically I could return error from this function, but then it would need to be passed up to the `BuildTable` method (3 levels to top). Given the fact that the only error case is when a dev makes a mistake, I don't think this is reasonable to do.

Also, as the function name says `IfShould`, that means IMHO it's ok to not do anything if the `routeMap` is nil. That's why I went with such approach.

I'm open for alternative ideas though 🙂

Reply:
Personally, I think it might be too defensive and if a map is nil, when it's not supposed to be, the runtime will never tell you.

You'll end in a false sense security. But ... def not a blocker. 👍

Reply:
> the runtime will never tell you.

I understand your point of view and I agree - yes, that's the downside of this if 👍 The upside is we'll avoid some panic which will crash the app. I came up with another possiblity: WDYT about converting these functions to methods and then use debug log to notify that the function was skipped because the map is nil?